MDANSE/Src/MDANSE/Framework/Jobs/CurrentCorrelationFunction.py

```diff
@@ -451,7 +453,8 @@ def run_step(self, index: int):
                     curr = np.einsum(
                         "ik,ij->ikj",
                         veloc,
-                        np.exp(1j * np.dot(coords, qVectors)),
+                        np.exp(1j * np.dot(coords, qVectors))
+                        * np.sqrt(qvec_weights[None, :]),
                     )
```

Comment:
Looks correct. No need for the new axis, could also do this.
```suggestion
                    curr = np.einsum(
                        "ik,ij,j->ikj",
                        veloc,
                        np.exp(1j * np.dot(coords, qVectors)),
                        np.sqrt(qvec_weights),
                    )
```